librz/arch/isa/msp430/msp430_il.c

```diff
@@ -0,0 +1,521 @@
+// SPDX-FileCopyrightText: 2024 Mostafa Mahmoud <ubermenchun@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <msp430/msp430_il.h>
+
+// ``The 16-bit program counter (PC/R0) points to the next instruction to be executed.``
+#define PC_SIZE 16
+// ``The low byte of a word is always an even address. The high byte is at the next odd address.
+//   For example, if a data word is located at address xxx4h, then the low byte of that data word
+//   is located at address xxx4h, and the high byte of that word is located at address xxx5h.``
+#define IS_BIG_ENDIAN false
+// implied by the width of the PC and other registers (which are used as pointers in the relevant addressing modes)
```

Comment:
Just reacently learned, that you can make those `doxygen` comments as well!

```
/**
 * \brief ...
 */
```

Might be usefult here

Reply:
Also, please `#undef` them at the end of the file. The names are very generic and might be used somewhere else.

Reply:
Hmmm, the generic names are probably not a problem because that's a `.c ` so it will never be included anywhere.

I could simply prefix a `MSP430_` to each one of them but it just feels redundant when the entire file is called `msp430_il.c` so of course any constant defined in it is about MSP430.

Reply:
>  so it will never be included anywhere.

ah yes, true.